crates/client-api/src/routes/subscribe.rs

```diff
@@ -74,10 +76,25 @@ where
 
     let db_address = name_or_address.resolve(&ctx).await?.into();
 
-    let (res, ws_upgrade, protocol) =
-        ws.select_protocol([(BIN_PROTOCOL, Protocol::Binary), (TEXT_PROTOCOL, Protocol::Text)]);
-
-    let protocol = protocol.ok_or((StatusCode::BAD_REQUEST, "no valid protocol selected"))?;
+    let (res, ws_upgrade, encoding) = ws.select_protocol([
+        (BIN_PROTOCOL, ProtocolEncoding::Binary),
+        (TEXT_PROTOCOL, ProtocolEncoding::Text),
+    ]);
+
+    let encoding = encoding.ok_or((StatusCode::BAD_REQUEST, "no valid protocol encoding selected"))?;
+    let protocol = Protocol {
+        encoding,
+        binary_compression: match std::env::var("SPACETIMEDB_COMPRESSION")
+            .map_or(String::from("default"), |s| s.to_lowercase())
+            .as_str()
+        {
+            "default" | "1" | "true" => ProtocolCompression::Brotli,
+            "no" | "0" | "false" => ProtocolCompression::None,
+            "none" => ProtocolCompression::None,
+            "brotli" => ProtocolCompression::Brotli,
+            _ => Err("Unhandled SPACETIMEDB_COMPRESSION value")?,
```

Comment:
Drive by comment, feel free to ignore.

Is the main purpose of this env var for testing? An env var might be overkill unless we think it will see use outside of testing.